### PR TITLE
[Merged by Bors] - feat: the monoidal structure on the category of presheaves of modules

### PR DESCRIPTION
**Pull request by joelriou**
This contribution was created as part of the AIM workshop "Formalizing algebraic geometry" in June 2024.

Co-authored-by: Dagur Asgeirsson <dagurtomas@gmail.com> 
Co-authored-by: Jack McKoen <mckoen@ualberta.ca> 

---

Following the refactor #16667, this is a cleaner attempt as compared to #14127.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Category/ModuleCat/Presheaf/Monoidal.lean [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    (by intro a m₁ m₂; dsimp; erw [M₁.map_smul]; rfl)
+    (by intro m₁ m₂ m₂'; dsimp; rw [map_add, TensorProduct.tmul_add])
+    (by intro a m₁ m₂; dsimp; erw [M₂.map_smul, TensorProduct.tmul_smul (r := R.map f a)]; rfl)
+
+/-- The tensor product of two presheaves of modules. -/
+@[simps obj]
+noncomputable def tensorObj : PresheafOfModules (R ⋙ forget₂ _ _) where
+  obj X := M₁.obj X ⊗ M₂.obj X
+  map {X Y} f := tensorObjMap M₁ M₂ f
+  map_id X := ModuleCat.MonoidalCategory.tensor_ext (by
```
Comment (dagurtomas) [inline Mathlib/Algebra/Category/ModuleCat/Presheaf/Monoidal.lean:53]:
```suggestion
  map f := tensorObjMap M₁ M₂ f
  map_id X := ModuleCat.MonoidalCategory.tensor_ext (by
```

**Review by dagurtomas — APPROVED** [review]
LGTM, thanks!

**Comment by dagurtomas** [comment]
I hope this is ok even though I contributed to the initial version of this:

maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**